management/src/commands/update-identity-from-user.ts

```diff
@@ -1,11 +1,19 @@
+import { ConnectUnreachableError } from "../errors";
 import { getIdentity } from "../queries/get-identity";
 import { ManagementCredentials } from "../types";
 import { getIdentityType } from "../utils/get-identity-type";
 import { addIdentityToUser } from "./add-identity-to-user";
 import { markIdentityAsPrimary } from "./mark-identity-as-primary";
 import { removeIdentityFromUser } from "./remove-identity-from-user";
 
-async function updateIdentityFromUser(
+const MAX_DELAY = 1000;
+
+async function delay(retryCount: number): Promise<void> {
+  const waitTime = Math.min(Math.pow(retryCount, 2) * 100, MAX_DELAY);
+  return new Promise((resolve) => setTimeout(resolve, waitTime));
+}
```

Comment:
That's a detail, but we don't need `async` there

```suggestion
function delay(retryCount: number): Promise<void> {
  const waitTime = Math.min(Math.pow(retryCount, 2) * 100, MAX_DELAY);
  return new Promise((resolve) => setTimeout(resolve, waitTime));
}
```